numpy/core/src/multiarray/datetime.c

```diff
@@ -1761,6 +1762,9 @@ parse_datetime_unit_from_string(char *str, Py_ssize_t len, char *metastr)
                 return NPY_FR_as;
         }
     }
+    else if (allow_generic && len == 7 && !strncmp(str, "generic", 7)) {
```

Comment:
Since it allows something that was an error before and never worked (I guess), we normally would never consider it a bugfix I think. Unless there is some other very good reason in this specific case?